### DISCUSSION TITLE
ClearType and vertical lines

### DISCUSSION THREAD

**Discussion by maa7** [rsms/inter#598] [General]
Hello,
As everyone knows, Cleartype's sub-pixel rendering improves the smoothing of slanted and curved parts of glyphs.
But as you can see in the screenshot, sub-pixel rendering is applied even for vertical glyph lines (see E, F, H, I, L, T) (for a unknown reason)(colored stripes on the right and left). Obviously a black and white border (like the horizontal lines) would be better.
Anyone know why this is being done?
![inter_chrome_4x](https://github.com/rsms/inter/assets/45234995/e553e739-a76d-4635-b0fc-be0a24b2783a)

**Comment by kenmcd** [comment]
> Anyone know why this is being done?

Ah, because that is the way ClearType works.
And that is affected by the resolution of your display, the size you have selected for viewing the text, the ClearType tuning settings you have on your Windows machine (if any), and by hinting in the font itself (if any and how much).

You can affect the ClearType rendering using the Windows built-in ClearType tuner.

You can also use free tools such as the Better ClearType Tuner.
https://github.com/bp2008/BetterClearTypeTuner 

You did not mention which version of the fonts you have installed.
TTF fonts can and do often include some TrueType hinting.
But the more elaborate the hinting, the bigger the font files.
So usually there is some general hinting done using automated tools.
That does not provide perfect results for all users, all display resolutions, and all font text display sizes.

Sometimes fonts can be extensively manually hinted (like the Microsoft ClearType collection fonts), but that is time-consuming, an expensive process, and done by experts in hinting, using horrible tools.
And it still does not have perfect results for all users, all resolutions, all text sizes.

You can hint the fonts yourself to possibly get the results you want, for the display you have, and for the text sizes you use - that way it is custom just for you.
Most fonts that are hinted use the free TTFAutohint tool to hint the fonts.
https://freetype.org/ttfautohint/doc/ttfautohint.html
There are settings in that free tool which _may_ get you the perfect black vertical stems you desire.

Some people like the way macOS renders type.
It just hammers the fonts into the grid. Nice and black.
The free MacType tool can give you this rendering on Windows.
https://github.com/snowie2000/mactype

ClearType attempts to minimize the visual distortion caused by forcing font outlines into a pixel grid. There are compromises.

**Comment by maa7** [comment]
> Ah, because that is the way ClearType works.

My question is why is that


> You did not mention which version of the fonts you have installed.
> TTF fonts can and do often include some TrueType hinting.

This is Inter without hinting: https://rsms.me/inter/lab/var.html
